app/containers/EmojiPicker/index.tsx

```diff
@@ -1,204 +1,187 @@
-	customEmojis: ICustomEmojis;
-	style?: StyleProp<ImageStyle>;
-	theme: TSupportedThemes;
-	onEmojiSelected: (emoji: string, shortname?: string) => void;
-	tabEmojiStyle?: StyleProp<TextStyle>;
-}
-
-interface IEmojiPickerState {
-	frequentlyUsed: (string | { content?: string; extension?: string; isCustom: boolean })[];
-	customEmojis: any;
-	show: boolean;
-	width: number | null;
-}
-
-class EmojiPicker extends Component<IEmojiPickerProps, IEmojiPickerState> {
-	constructor(props: IEmojiPickerProps) {
-		super(props);
-		const customEmojis = Object.keys(props.customEmojis)
-			.filter(item => item === props.customEmojis[item].name)
+import { useTheme } from '../../theme';
+import { IEmoji, ICustomEmojis, TFrequentlyUsedEmojiModel } from '../../definitions';
+import { useAppSelector } from '../../lib/hooks';
+import { IEmojiPickerProps, EventTypes } from './interfaces';
+
+const EmojiPicker = React.memo(
+	({
+		baseUrl,
+		onItemClicked,
+		tabEmojiStyle,
+		isEmojiKeyboard = false,
+		searching = false,
+		searchedEmojis = []
+	}: IEmojiPickerProps) => {
+		const [frequentlyUsed, setFrequentlyUsed] = useState<IEmoji[]>([]);
+		const [show, setShow] = useState(false);
+		const [width, setWidth] = useState(null);
+		const { colors } = useTheme();
+
+		const allCustomEmojis: ICustomEmojis = useAppSelector(state => state.customEmojis);
+		const customEmojis = Object.keys(allCustomEmojis)
+			.filter(item => item === allCustomEmojis[item].name)
 			.map(item => ({
-				content: props.customEmojis[item].name,
-				extension: props.customEmojis[item].extension,
+				content: allCustomEmojis[item].name,
+				extension: allCustomEmojis[item].extension,
 				isCustom: true
 			}));
-		this.state = {
-			frequentlyUsed: [],
-			customEmojis,
-			show: false,
-			width: null
-		};
-	}
-
-	async componentDidMount() {
-		await this.updateFrequentlyUsed();
-		this.setState({ show: true });
-	}
-
-	shouldComponentUpdate(nextProps: IEmojiPickerProps, nextState: IEmojiPickerState) {
-		const { frequentlyUsed, show, width } = this.state;
-		const { theme } = this.props;
-		if (nextProps.theme !== theme) {
-			return true;
-		}
-		if (nextState.show !== show) {
-			return true;
-		}
-		if (nextState.width !== width) {
-			return true;
-		}
-		if (!dequal(nextState.frequentlyUsed, frequentlyUsed)) {
-			return true;
-		}
-		return false;
-	}
 
-	onEmojiSelected = (emoji: IEmoji) => {
-		try {
-			const { onEmojiSelected } = this.props;
-			if (emoji.isCustom) {
-				this._addFrequentlyUsed({
-					content: emoji.content,
-					extension: emoji.extension,
-					isCustom: true
-				});
-				onEmojiSelected(`:${emoji.content}:`);
-			} else {
-				const content = emoji;
-				this._addFrequentlyUsed({ content, isCustom: false });
-				const shortname = `:${emoji}:`;
-				onEmojiSelected(shortnameToUnicode(shortname), shortname);
+		useEffect(() => {
+			const init = async () => {
+				await updateFrequentlyUsed();
+				setShow(true);
+			};
+			init();
+		}, []);
```

Comment:
Can you turn this into a custom hook?
You would extract most of this logic away from the component.